### ISSUE TITLE
Display settings

### ISSUE THREAD

**Issue by Nezz** [theneweinstein/pysomneo#13]
My wake up light has two display settings in the app:
- Display always on
- Display brightness

Would it be possible to control these? I'd like enable always on during the day, but disable it during the night from Home Assistant.

**Comment by Frank071** [comment]
These probably are ```brght``` (brightness) and ```dspon``` (true = always on) from the ```wusts``` call

**Comment by Nezz** [comment]
Thanks for adding support @theneweinstein. I gave it a try in home assistant. I noticed that when setting both the brightness and always on from the same automation, the first setting gets lost. Example:

```
  - service: number.set_value
    metadata: {}
    data:
      value: "1"
    target:
      entity_id: number.somneo_display_brightness
  - service: switch.turn_off
    metadata: {}
    data: {}
    target:
      entity_id: switch.somneo_display_always_on
```

When the brightness is initially set to a different value, running this automation results in the newly set brightness being set for a split second, which gets reverted back to the initial setting by the always on switch.

Adding a two second delay between the two calls fixes the issue.

**Comment by VictorLabeille** [comment]
Disclaimer : My Somneo is a HF3671/01 model swverwifi 2.2.5 with the firmware pkgver 30406 so the results I got (by testing with a LAN host) and my conclusions may not be the same on other Somneos.

As previously stated the original request was fulfilled with the function set_display(state, brightness), how ever it looks like there's a typo in the docstrings. 
They're saying `brightness: Display brightness level (0-255)` however my somneo only accepts values from 1 to 6. 
Here are the tests I did :

| Written value | Response | Read value |
| ------------- | -------- | ---------- |
| 1             | `200`    | 1          |
| 6             | `200`    | 6          |
| 0             | `422`    | unchanged  |
| 7             | `422`    | unchanged  |
| 128           | `422`    | unchanged  |

While `ltlvl` uses 0-255 in `_update_light_data`, `brght` is read and transmitted as-is in `_update_alarm_status`. So the docstring mentioned above should say 1-6 ?

I have looked into the problem mentionned by @Nezz :
For your problem where setting both the brightness and always on from the same automation losses the first setting, fixed by a 2 seconds delay. It's not a network concurrency problem but a client-side cache, it's visible in the old code from june 2024 :
```python
payload['dspon'] = state if state != None else self.data['display_always_on']
payload['brght'] = brightness if brightness else self.data['display_brightness']
self.alarm_status = self._put('wusts', payload = payload)
```
So `set_display` always send both fields, the one we want to set and the one from cache and this last one wasn't being refreshed after the PUT. The second call was just rewritting the original value.
Here is what I reproduced on my Somneo with june 2024's logic :

| Call                        | `PUT` sent                     | State read back  |
| --------------------------- | ------------------------------ | ---------------- |
| `set_display(brightness=4)` | `{"dspon": false, "brght": 4}` | `brght=4`        |
| `set_display(state=True)`   | `{"dspon": true, "brght": 1}`  | `brght=1` — lost |

As opposed to these measurments, I tested with pysomneo 5.0.6 and it works well.
Your 2 seconds delay was probably working because it gave the HA coordinator time to refresh the cache.

**TLDR :** This bug is already corrected since commit `3fc2c33` (september 20th, 2025). If not for the small docstring typo, the issue can be closed :)

Btw it looks like a refactoring of the codebase to be async is on its way in the branch `ai-improvements` so fixing the docstring in master may be useless. How do you want me to contribute @theneweinstein ?
